File: gpt-3.5/template_only/acc_memcpy_from_device.c
#include "acc_testsuite.h"

#ifndef T1
// T1:runtime,data,executable-data,construct-independent,V:2.0-2.7
int test1(){
    int err = 0;
    srand(SEED);

    // Declare and initialize variables
    int n = 100;
    int *data = (int*)malloc(n * sizeof(int));
    int *devData;

    // Allocate memory on the device
    #pragma acc enter data create(data[0:n])

    // Initialize data on the device
    #pragma acc parallel loop present(data[0:n])
    for (int i = 0; i < n; i++){
        data[i] = rand();
    }

    // Allocate memory on the host
    devData = (int*)malloc(n * sizeof(int));

    // Copy data from the device to the host using acc_memcpy_from_device
    #pragma acc host_data use_device(data)
    {
        #pragma acc memcpy from(device:data[0:n]) to(host:devData[0:n])
    }

    // Verify the data on the host matches the data on the device
    for (int i = 0; i < n; i++){
        if (devData[i] != data[i]){
            err = 1;
            break;
        }
    }

    // Free memory on the device and host
    #pragma acc exit data delete(data[0:n])
    free(data);
    free(devData);

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;

#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif

    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/acc_memcpy_from_device.f90
      !T1:acc memcpy from device,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=4), DIMENSION(N) :: a, a_host, a_dev
        INTEGER :: i

        ! Initialize host array
        DO i = 1, N
          a_host(i) = REAL(i)
        END DO

        ! Copy host array to device
        CALL acc_copyin(a_host, N)

        ! Copy device array to host
        CALL acc_memcpy_from_device(a_dev, C_LOC(a_host), N*4)

        ! Compare host and device arrays
        DO i = 1, N
          IF (a_host(i) .ne. a_dev(i)) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM